File: 1323-maximum-69-number/1323-maximum-69-number.cpp
#include <string> 
class Solution {
public:
    int maximum69Number (int num) {
        int i = 0;
        auto s = std::to_string(num);
        int n = s.length();
        while ((i < n) & (s[i] == '9')) {
            i++;
        }
        if (i == n) {
            return num;  // only 9s
        }
        string s2 = s.substr(0, i) + "9" + s.substr(i+1);
        return stoi(s2);  // replace the leftmost 6 by a 9
    }
};

File: 1323-maximum-69-number/1323-maximum-69-number.py
class Solution:
    def maximum69Number(self, num: int) -> int:
        i = 0
        s = str(num)
        n = len(s)
        while (i < n) and (s[i] == "9"):
            i += 1
        if (i == n):
            return num  # only 9s
        return int(s[:i]+"9"+s[i+1:])  # replace the leftmost 6 by a 9